pip_audit/_dependency_source/resolvelib/resolvelib.py

```diff
@@ -82,6 +90,17 @@ def resolve(self, req: Requirement) -> list[Dependency]:
         except HTTPError as e:
             raise ResolveLibResolverError("failed to resolve dependencies") from e
         for name, candidate in result.mapping.items():
+            # Check hash validity
+            if req_hashes:
+                for alg, hashes in req_hashes[name].items():
+                    for h in hashes:
+                        if h == candidate.dist_hashes[alg]:
+                            break
+                    else:
+                        raise ResolveLibResolverError(
+                            f"Mismatched hash for {name} ({candidate.version}): listed {h} of type "
+                            f"{alg} could not be found"
+                        )
```

Comment:
This kind of boilerplate is definitely a good argument for having `RequirementHashes` be a full model with a `match` or similar method 🙂

Reply:
[698d897](https://github.com/pypa/pip-audit/pull/462/commits/698d897b04186c909c0cd30a97e504f4b7703277)